### PR TITLE
fix:[CI-13143] fix for deleteDir conversion logic

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
* The deleteDir step script was not correctly formatted and caused problems when imported into the Harness pipeline.
* The dir function was turning into ```rm -rf``` command.

### CODE REVIEW

**Review by hemanthmantri — APPROVED** [review]
Isn't there a UnitTest to update?

**Comment by Fernando-Dourado** [comment]
> Isn't there a UnitTest to update?

You are right, the actual unit tests don't look healthy. Let me review them.

File: convert/jenkinsjson/json/deletedir.go [hunk 1]
```diff
@@ -29,12 +25,12 @@ func ConvertDeleteDir(node Node, variables map[string]string) *harness.Step {
...
 	dirPath := node.ParameterMap["path"].(string)
 	step := &harness.Step{
-		Name: "Deletingdir",
+		Name: "Dir",
 		Id:   SanitizeForId(node.SpanName, node.SpanId),
 		Type: "script",
 		Spec: &harness.StepExec{
 			Shell: "sh",
-			Run:   fmt.Sprintf("rm -rf %s", dirPath),
+			Run:   fmt.Sprintf("mkdir %s && cd %s", dirPath, dirPath),
```
Comment (jamie-harness) [inline convert/jenkinsjson/json/deletedir.go:33]: we might want mkdir -p here

**Comment by jamie-harness** [comment]
Let's document/mention somewhere this solution should work for most cases. But will behave a bit differently for nested dir command on jenkins side

**Comment by Fernando-Dourado** [comment]
> Let's document/mention somewhere this solution should work for most cases. But will behave a bit differently for nested dir command on jenkins side

Where do you recommend to put that information?

File: convert/jenkinsjson/json/deletedir.go (hunk 1 above)
Comment (Fernando-Dourado) [reply convert/jenkinsjson/json/deletedir.go]: Done

**Comment by jamie-harness** [comment]
> > Let's document/mention somewhere this solution should work for most cases. But will behave a bit differently for nested dir command on jenkins side
> 
> Where do you recommend to put that information?

Please talk to Prasad. We can decide on something. I have this current limitation doc in general: https://harness.atlassian.net/wiki/spaces/CI1/pages/22010724438/Internal+Limitation+Documentation

Let's put in the Jira comment while we wait for a reply

**Review by jamie-harness — APPROVED** [review]

**Pull request merged**